controller/sync.go

```diff
@@ -578,13 +578,13 @@ func (sc *syncContext) pruneObject(liveObj *unstructured.Unstructured, prune, dr
 }
 
 func (sc *syncContext) hasCRDOfGroupKind(group string, kind string) bool {
-	for _, res := range sc.compareResult.managedResources {
-		if res.Target != nil && kube.IsCRD(res.Target) {
-			crdGroup, ok, err := unstructured.NestedString(res.Target.Object, "spec", "group")
+	for _, obj := range sc.compareResult.objs() {
```

Comment:
Looks like now method returns true if either live or target object is CRD. This is incorrect if target is nil and live is not nil. In this case crd won't be created so result should be false.

Reply:
I'll revert this.